umake/frameworks/ide.py

```diff
@@ -473,3 +474,115 @@ def post_install(self):
         if not self.was_in_arduino_group:
             UI.delayed_display(DisplayMessage(_("You need to logout and login again for your installation to work")))
+            
+class BaseNetBeans(umake.frameworks.baseinstaller.BaseInstaller):
+    """The base for all Netbeans installers."""
+    
+    BASE_URL = "http://download.netbeans.org/netbeans/"
+    executable = "nb/netbeans"
+
+    def __init__(self, category, flavour=""):
+        """ The constructror.
+        @param category The IDE category.
+        @param flavour The Netbeans flavour (plugins bundled).
+        """
+        #add a separator to the string, like -cpp
+        if flavour != "":
+            flavour = '-' + flavour
+        self.flavour = flavour
+        
+        super().__init__(name="Netbeans",
+                         description=_("Netbeans IDE"),
+                         category=category, 
+                         only_on_archs=['i386', 'amd64'], #FIXME add arm support
+                         download_page=None,
+                         dir_to_decompress_in_tarball="*", #FIXME windows and mac files not required
+                         desktop_filename="netbeans{}.desktop".format(flavour),
+                         #FIXME support for oracle java and multiple java versions
+                         packages_requirements=['openjdk-7-jdk', 'jayatana']) 
+        
+    def search_url_string(self, url_file, version="", flavour=""):
+        """Search the url appendix for the searched IDE.
+        @param url_file The downloaded file.
+        @param version The version of the IDE.
+        @param flavour The flavour (plugins bundled) of the IDE.
+        @return A string in the form: suffix, 8digit number, md5sum, languages
+        """
+        preg = re.compile('add_file\("zip/netbeans-' + version\
+                        + '-[0-9]{12}' + flavour + '.zip"')
+        for line in url_file:
+            for line in url_file.split("\n"):
+                if preg.match(line):
+                    #Clean up the string from js (it's a function call)
+                    line = line.replace("add_file(", "").replace(");", "")
+                    line = line.replace('"', "")
+                    return line
+                    
+    def get_latest_version(self):
+        """Obtain the latest version of the IDE"""
+        url_version = urlopen("https://netbeans.org/downloads/zip.html")
+        url_version_str = url_version.read().decode('utf-8')
+        preg = re.compile(".*/images_www/v6/download/.*")
+        for line in url_version_str.split("\n"):
+            if preg.match(line):
+                line = line.replace("var PAGE_ARTIFACTS_LOCATION = \"/images" + \
+                                    "_www/v6/download/", "").replace("/\";", "")
+                return line.strip()
+        #Fallback
+        logger.error("Could not determine latest version, using fallback...")
+        return "8.0.2"
```

Comment:
I would prefer such hardcoded fallbacks to be a static property of the class on top, to ensure we don't have to read through the class implementation to change it.

So something next to BASE_URL, like FALLBACK_VERSION="8.0.2"